### PR TITLE
Bug fixes to proximals

### PR DESCRIPTION
**Pull request by adler-j**
Fixed some bugs with proximals that our largescale tests showed.

### CODE REVIEW

File: odl/solvers/nonsmooth/proximal_operators.py [hunk 1]
```diff
@@ -724,8 +724,10 @@ def __init__(self, sigma):
         def _call(self, x, out):
             """Apply the operator to ``x`` and stores the result in ``out``."""
 
+            eps = np.finfo(float).resolution * 10
```
Comment (kohr-h) [inline odl/solvers/nonsmooth/proximal_operators.py]: Why not `self.domain.dtype`?

File: odl/solvers/functional/default_functionals.py [hunk 2]
```diff
@@ -578,6 +607,10 @@ def __init__(self, space):
         """
         IndicatorBox.__init__(self, space, lower=0, upper=None)
 
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        return '{}()'.format(self.__class__.__name__)
```
Comment (kohr-h) [inline odl/solvers/functional/default_functionals.py]: `self.domain`

File: odl/test/largescale/solvers/nonsmooth/proximal_operator_slow_test.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+def test_proximal_defintion(functional, stepsize):
+    """Test the defintion of the proximal:
+
+        prox[f](x) = argmin_y {f(y) + 1/2 ||x-y||^2}
+
+    Hence we expect for all x in the domain of the proximal
+
+        x* = prox[f](x)
+
+        f(x*) + 1/2 ||x-x*||^2 < f(y) + 1/2 ||x-y||^2
```
Comment (kohr-h) [inline odl/test/largescale/solvers/nonsmooth/proximal_operator_slow_test.py]: Nice description!
I'd write "<=" just to be on the safe side.

File: odl/test/largescale/solvers/nonsmooth/proximal_operator_slow_test.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        f_y = proximal_objective(stepsize * functional, x, y)
+
+        if not f_prox_x <= f_y + EPS:
+            print(functional, x, y, f_prox_x, f_y)
+
+        assert f_prox_x <= f_y + EPS
+
+
+def test_proximal_cconj_kl_cross_entropy_solving_opt_problem():
+    """Test for proximal operator of conjguate of 2nd kind KL-divergecen.
```
Comment (kohr-h) [inline odl/test/largescale/solvers/nonsmooth/proximal_operator_slow_test.py]: divergence

File: odl/solvers/nonsmooth/proximal_operators.py (hunk 1 above)
Comment (adler-j) [inline odl/solvers/nonsmooth/proximal_operators.py]: Not guaranteed to exist but i guess we can go for it.

Comment (adler-j) [inline odl/solvers/nonsmooth/proximal_operators.py]: I'll getattr:

```
getattr(self.domain, 'dtype', float)
```

**Comment by adler-j** [comment]
Merge after CI

**Pull request merged**